PocketKit/Sources/PocketKit/CarPlay/CarPlayTemplateManager.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  CarPlayTemplateManager.swift
+//  PKTListen
+//
+//  Created by Daniel Brooks on 4/17/23.
+//  Copyright © 2023 PKT. All rights reserved.
+//
+
+import CarPlay
+import Foundation
+import os
+import MediaPlayer
+import PKTListen
+
+class CarPlayTemplateManager: NSObject {
+    /// A reference to the CPInterfaceController that passes in after connecting to CarPlay.
+    private var carplayInterfaceController: CPInterfaceController?
+
+    /// The CarPlay session configuation contains information on restrictions for the specified interface.
+    var sessionConfiguration: CPSessionConfiguration!
+
+    /// The observer of the Now Playing item changes.
+    var nowPlayingItemObserver: NSObjectProtocol?
+
+    /// The observer of the playback state changes.
+    var playbackObserver: NSObjectProtocol?
+
+    /// PKT Listen Configuration
+    var saves: PKTListenConfiguration
+
+    var archive: PKTListenConfiguration
+
+    // TODO: is there a better more abstract init here so we can move CarPlayTemplate manager into PKTListen?
+    init(saves: PKTListenConfiguration, archive: PKTListenConfiguration) {
+        self.saves = saves
+        self.archive = archive
+    }
+
+    func connect(_ interfaceController: CPInterfaceController) {
+        carplayInterfaceController = interfaceController
+        carplayInterfaceController!.delegate = self
+        sessionConfiguration = CPSessionConfiguration(delegate: self)
+        addObservers()
+        carplayInterfaceController!.setRootTemplate(loggedInTemplate(), animated: true, completion: nil)
+    }
+
+    /// Called when CarPlay disconnects.
+    func disconnect() {
+        nowPlayingItemObserver = nil
+        playbackObserver = nil
+        MPMusicPlayerController.applicationQueuePlayer.pause()
+    }
+}
+
+extension CarPlayTemplateManager {
+
+    private func loggedInTemplate() -> CPTabBarTemplate {
+        var tabTemplates = [CPTemplate]()
+        tabTemplates.append(savesTemplate())
+        tabTemplates.append(archiveTemplate())
+        tabTemplates.append(playlistsTemplate())
+        return CPTabBarTemplate(templates: tabTemplates)
+    }
+
+    private func savesTemplate() -> CPListTemplate {
+        let items: [CPListItem] = saves.source.list.compactMap({ item in
```

Comment:
@nzeltzer How do I get the raw PKTListenItems out of the PKTListenConfiguration class?